### ISSUE TITLE
AUR package search still inferior to the original

### ISSUE THREAD

**Issue by RubenKelevra** [fosskers/aura#923]
I've seen that Aura did big steps in proving the AUR search, but I did one today where the result is odd to me (as user). I understand, why there are differences as a programmer. However it makes usability pretty hard if the search is bad:

```
$ aura -As nexus mods

$ paru -Ss nexus mods
aur/dmodman-git r306.54a0274-1 [+0 ~0.00]
    TUI downloader & update checker for Nexusmods.com
aur/nexusmods-app-bin 0.5.3-2 [+2 ~1.17]
    A mod installer, creator and manager for all your popular games.
```

Hope you can improve this somehow without much effort :)

**Comment by fosskers** [comment]
[As mentioned in the guide](https://fosskers.github.io/aura/search.html), at the moment only full terms can be matched on.

**Comment by RubenKelevra** [comment]
Wondering if nexus and mods may be keywords?

Is it possible to read the keywords for packages via the API? Because as regular users you don't see them. Only as maintainer you got a field to fill them out.

**Comment by RubenKelevra** [comment]
I guess not. They really do substring matching. Weird.

```
$ paru -Ss nexu mo
aur/nexusmods-app-bin 0.5.3-2 [+3 ~2.10] [Installed]
    A mod installer, creator and manager for all your popular games.
aur/dmodman-git r306.54a0274-1 [+0 ~0.00]
    TUI downloader & update checker for Nexusmods.com
```

**Comment by fosskers** [comment]
It's probably just calling the AUR directly, which does substring matching internally.

**Comment by RubenKelevra** [comment]
Yeah, I was just wondering. That's not a very performant operation, weird that they do this on the AUR database :)

**Comment by fosskers** [comment]
Some databases do support such full-text search, but running a database was something I wanted to avoid.

**Comment by TheMatten** [comment]
+1 for this being very impractical - if user doesn't know exact name of the package, it's easy to miss some characters, e.g. in shortcuts or singular vs. plural.

**Comment by RubenKelevra** [comment]
> Some databases do support such full-text search, but running a database was something I wanted to avoid.

How about not running a database and instead have aura download the database and run the search locally?

Would be much faster, available offline, no additional server cost and you could provide simple update files, say compressed sql files with the delta, so aura could catch up and don't have to re-download the whole database on each update.

If you don't want to add something like sqlite, there's probably plenty of in memory databases available for rust. So you would simply provide a list with the AUR metadata and load it in the in-memory database on a query, before searching it.

zstd can create the tiny delta updates for the file with its `--patch-from=` feature.

So you would just offer some tiny static files to download, probably packing them from time to time into tar files for quicker downloads.

**Comment by fosskers** [comment]
Interesting idea but the size would be prohibitive, especially to keep updated for all users.

**Comment by RubenKelevra** [comment]
@fosskers You're sure? There's only 87984 packages in the AUR. So one sentence, a couple of keywords, a version number and a name as list wouldn't weight more than a Megabyte compressed I would have guessed.

**Comment by fosskers** [comment]
You'd be surprised. Keeping it updated for everyone wouldn't be worth it.
